### DISCUSSION TITLE
Protect my own yaml file from deleting after upgrade tuya local

### DISCUSSION THREAD

**Discussion by Nevynnyi** [make-all/tuya-local#2300] [Q&A]
Hi to all!

I have three device and they are not supported from tuya local. I am not developer, but i crate my own yaml files in devise and they are worked. But every time, after upgrade tuya locale - they deleted. How can i protect them?

My device:
atorch_gr2pws_breaker
cronos_heater_gph-d-da
earu_1p_breaker

I whrite to add them and give cod.
Now i use "chattr +i" command in my server, and looking for result.
Maybe you can suggest another way that I don’t know or haven’t found.

**Comment by Trzinka** [comment]
> Hi to all!
> 
> I have three device and they are not supported from tuya local. I am not developer, but i crate my own yaml files in devise and they are worked. But every time, after upgrade tuya locale - they deleted. How can i protect them?
> 
> My device: atorch_gr2pws_breaker cronos_heater_gph-d-da earu_1p_breaker
> 
> I whrite to add them and give cod. Now i use "chattr +i" command in my server, and looking for result. Maybe you can suggest another way that I don’t know or haven’t found.

### Posted under Issues:

`New device supported`

Add code and Your code will be reviewed and added to the new version shortly.

**Comment by Nevynnyi** [comment]
Thx! 🤗🙏

**Comment by rthsw** [comment]
Hi, i have the same issue with an Earu breaker... where can i add my modifications or my requests?
The File is earu_dpn_circuitbreaker.yaml

There is one mistake: The earu dpn breaker reports watts in the scale of 10.000, not 10.

Original:
- entity: sensor
    category: diagnostic
    class: power
    dps:
      - id: 19
        name: sensor
        type: integer
        unit: kW
        class: measurement
        force: true
        mapping:
          - scale: 10
Correct:
- entity: sensor
    category: diagnostic
    class: power
    dps:
      - id: 19
        name: sensor
        type: integer
        unit: kW
        class: measurement
        force: true
        mapping:
          - scale: 10000
          
And the breaker delivers more entities:
          
      "9": 0, //Timer as INT, Hour * 3600 + Minute * 60 + Seconds. 7 Hours, 6 Minutes, 3 Seconds = 25563. 0 = No timer active
      "38": "on", //Relay Status on power ON. Options Off, ON, Memory
      "40": "relay" //State of the LED. Option relay, position, none, ON
      "47": 41 //Temperatur in °C

I think the Alarm settings & Tresholds (Overvoltage, Over temperature, Under Voltage etc) are coded in the Flag 48 and 49
      "48": "BQAAQgcBABU=",
      "49": "AQEABgMBAOoEAQBz"

But i could not figure out now how.
 If someone help me how to change the YAML file according to my Test-File research, i would be happy (and also try to figure out the mappings).

**Reply by make-all** [reply]
To figure out the mappings, the best way is to get the full documentation available from tuya using the Query Things Data Model API call through the developer portal. Without that, it is just guesswork.
